pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -149,34 +268,431 @@ func (c *Check) instanceConfigure(data integration.Data) error {
 			}
 		}
 	}
+	tagByLabel, found := unmarshalledInstanceConfig["tag_by_label"]
+	if tagByLabel, ok := tagByLabel.(bool); found && ok {
+		c.cfg.tagByLabel = tagByLabel
+	}
+	useLsblk, found := unmarshalledInstanceConfig["use_lsblk"]
+	if useLsblk, ok := useLsblk.(bool); found && ok {
+		c.cfg.useLsblk = useLsblk
+	}
+	blkidCacheFile, found := unmarshalledInstanceConfig["blkid_cache_file"]
+	if blkidCacheFile, ok := blkidCacheFile.(string); found && ok {
+		c.cfg.blkidCacheFile = blkidCacheFile
+	}
+	if c.cfg.tagByLabel && c.cfg.useLsblk && c.cfg.blkidCacheFile != "" {
+		return errors.New("only one of 'use_lsblk' and 'blkid_cache_file' can be set at the same time")
+	}
+	serviceCheckRw, found := unmarshalledInstanceConfig["service_check_rw"]
+	if serviceCheckRw, ok := serviceCheckRw.(bool); found && ok {
+		c.cfg.serviceCheckRw = serviceCheckRw
+	}
+	c.configureCreateMounts(unmarshalledInstanceConfig)
+	return nil
+}
 
+func (c *Check) configureExcludeDevice(instanceConfig map[interface{}]interface{}, initConfig map[interface{}]interface{}) error {
+	for _, key := range []string{"device_global_exclude", "device_global_blacklist"} {
+		if deviceExclude, ok := initConfig[key].([]interface{}); ok {
+			for _, val := range deviceExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+				}
+			}
+		}
+	}
+	for _, key := range []string{"device_exclude", "device_blacklist", "excluded_disks"} {
+		if deviceExclude, ok := instanceConfig[key].([]interface{}); ok {
+			for _, val := range deviceExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
```

Comment:
The Python versions adds a $ at the end of the regex for `excluded_disks`
[DataDog/integrations-core@master/disk/datadog_checks/disk/disk.py#L368](https://github.com/DataDog/integrations-core/blob/master/disk/datadog_checks/disk/disk.py#L368)